### PR TITLE
CMN-1245 Add order by clause

### PR DESCRIPTION
**Pull request by if-loop69420**
## Summary
What's in the ticket and in the name

## What has been done? 
- Add tokens and syntaxkinds to support order by clauses in queries
- add parse_order_by_clause to parse_query
- write tests for order_by_clauses and insure previous tests still run
---

### CODE REVIEW

File: src/grammar/query.rs [hunk 1]
```diff
@@ -181,6 +185,31 @@ pub(crate) fn parse_where_clause(p: &mut Parser) {
...
 
+pub(crate) fn parse_order_by_clause(p: &mut Parser) {
+    p.start(SyntaxKind::OrderByClause);
+    p.expect(T![order]);
+    p.eat(T![siblings]);
+    p.expect(T![by]);
+    safe_loop!(p, {
+        if p.at(T![int_literal]) {
+            p.eat(T![int_literal]);
+        } else {
```
Comment (KieranKaelin) [inline src/grammar/query.rs:196]:
```suggestion
        if !p.eat(T![int_literal]) {
```

File: src/grammar/query.rs [hunk 2]
```diff
@@ -181,6 +185,31 @@ pub(crate) fn parse_where_clause(p: &mut Parser) {
...
+            parse_expr(p);
+        }
+        if [T![asc], T![desc]].contains(&p.current()) {
+            p.expect_one_of(&[T![asc], T![desc]]);
+        }
```
Comment (KieranKaelin) [inline src/grammar/query.rs:201]: There's a `p.eat_one_of` function for these use-cases :smile:

**Review by KieranKaelin — CHANGES_REQUESTED** [review]
Please also cover `NULLS FIRST` and int literals in the tests

File: src/grammar/query.rs [hunk 3]
```diff
@@ -28,6 +28,10 @@ pub(crate) fn parse_query(p: &mut Parser, expect_into_clause: bool) {
         _ => (), // No-op
     }
 
+    if p.current() == T![order] {
```
[the remark this answers is not in the document]
Comment (if-loop69420) [reply src/grammar/query.rs:31]: True

**Review by KieranKaelin — APPROVED** [review]

**Pull request merged**